pkg/controller/job/metrics/metrics.go

```diff
@@ -78,9 +78,10 @@ var (
 	//   result:          failed, succeeded
 	JobPodsFinished = metrics.NewCounterVec(
 		&metrics.CounterOpts{
-			Subsystem: JobControllerSubsystem,
-			Name:      "job_pods_finished_total",
-			Help:      "The number of finished Pods that are fully tracked",
+			Subsystem:      JobControllerSubsystem,
+			Name:           "job_pods_finished_total",
```

Comment:
Would `job_controller_pods_finished_total` be enough? or shall we keep the `jobs_` prefix @dgrisonnet